backend/src/db/cache_storage.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from enum import Enum
+from typing import Optional
+
+from redis import Redis
+
+from src.core.configs import (
+    CACHE_STORAGE_EXP,
+    CACHE_STORAGE_HOST,
+    CACHE_STORAGE_PASSWORD,
+    CACHE_STORAGE_PORT,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class CacheKeys(Enum):
+    THREAD = "thread:"
+    COMMENT = "thread-comment:"
+    ATTRACTION = "attraction:"
+    ATTRACTION_IMAGES = "attraction-images:"
+
+
+class CacheStorage:
+    def __init__(
+        self,
+        host: Optional[str] = None,
+        port: Optional[str] = None,
+        password: Optional[str] = None,
+        expiration: Optional[int] = None,
+    ):
+        self._host = host or CACHE_STORAGE_HOST
+        self._port = port or CACHE_STORAGE_PORT
+        self._password = password or CACHE_STORAGE_PASSWORD
+        self.cache = self._connect_to_data_store()
+
+        self.expiration_time: int = expiration or CACHE_STORAGE_EXP
+
+    def _connect_to_data_store(self) -> Redis:
+        try:
+            redis = Redis(
+                host=self._host,
+                port=self._port,
+                password=self._password,
+                decode_responses=True,
+            )
+            return redis
+        except Exception as e:
+            logger.error("Could not connect to redis server: %s" % e)
+            raise Exception(f"Could not connect to redis server: {e}")
```

Comment:
Podmień to aby nie była to metoda prywatna, abym można było sobie spokojnie testować stan redisa
np:
`import redis
r = cache.connect_to_data_store()
r.keys("*")
['user:2f6fe323-4ae0-4de4-a1e4-e27b0773170d']`